### PR TITLE
cli: add auth subcommand and fix subcommand options checks

### PR DESCRIPTION
**Pull request by unleashed**

### CODE REVIEW

File: src/cli/subcommands/auth.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            .help("Specify the service ID"))
+        .arg(Arg::with_name("service_token")
+            .short("t")
+            .long("service-token")
+            .takes_value(true)
+            .value_name("SERVICE_TOKEN")
+            .help("Specify the service token"))
+        .arg(Arg::with_name("provider_key")
+            .short("k")
+            .long("provider-kery")
```
Comment (davidor) [inline src/cli/subcommands/auth.rs]: typo

**Comment by davidor** [comment]
@unleashed I added a new commit to adapt the threescale.status() to your changes.
The PR looks good to me :+1:

File: src/threescale.rs [hunk 2]
```diff
@@ -11,6 +11,8 @@ use http::HttpClient;
 use service::Service;
 use service::credentials::Credentials;
 
+const STATUS_ENDPOINT: &'static str = "/status";
```
Comment (unleashed) [inline src/threescale.rs]: I think `'static` is not needed here, because, well, it can _only_ be `'static`.

Comment (davidor) [inline src/threescale.rs]: Right. Changed.

**Pull request merged**